searx/plugins/calculator.py

```diff
@@ -51,10 +55,31 @@ def _eval(node):
     raise TypeError(node)
 
 
+def timeout_func(timeout, func, *args, **kwargs):
```

Comment:
This seems like a useful utility method that could be used for any general func. Do you think this has the possibility to be used in other places in the project, and if so for it to belong in a separate shared file?

Reply:
I had already thought about it, but then decided against it: Forking a new process is a resource "expensive" operation and should remain the exception. Should we one day have the need elsewhere, we could then consider moving this function to another module such as `searx.utils`.